path_oidc.go

```diff
@@ -502,12 +502,10 @@ func validRedirect(uri string, allowed []string) bool {
 
 // parseMount attempts to extract the mount path from a redirect URI.
 func parseMount(redirectURI string) string {
-	parts := strings.Split(redirectURI, "/")
-
-	for i := 0; i+2 < len(parts); i++ {
-		if parts[i] == "v1" && parts[i+1] == "auth" {
-			return parts[i+2]
-		}
+	parts := strings.Split(redirectURI, "/v1/auth/")
```

Comment:
Nice.

Reply:
Actually...it doesn't work and I restore the old code and revised the test. We only want the immediate next element, not the rest of the path.